src/info/Info.ts

```diff
@@ -0,0 +1,137 @@
+        if (/Android/.test(navigator.userAgent)) {
+            return "Android";
+        } else if (/iPhone|iPad|iPod/.test(navigator.userAgent)) {
+            return "iOS";
+        } else if (platform.startsWith("Win")) {
+            return "Windows";
+        } else if (platform.startsWith("Mac")) {
+            return "MacOS";
+        } else if (platform.startsWith("Linux")) {
+            return "Linux";
+        } else {
+            return "Unknown";
+        }
+    }
+
+    public static GetSystemVersion(): string {
+        const userAgent = navigator.userAgent;
+        const match = userAgent.match(/(Windows NT|Mac OS X|Android|CPU iPhone OS|CPU OS|Linux) ([\d._]+)/);
+        if (match) {
+            return match[2].replace(/_/g, '.');
+        }
+        return "Unknown";
+    }
+
+    public static async GetSupportsWebXR(): Promise<boolean> {
+        return new Promise((resolve) => {
+            if (this._supportsWebXR !== null) {
+                resolve(this._supportsWebXR);
+                return;
+            }
+
+            if (!navigator.xr) {
+                this._supportsWebXR = false;
+                resolve(false);
+                return;
+            }
+            // Check if immersive-vr session mode is supported
+            navigator.xr.isSessionSupported('immersive-ar')
+                .then((supported) => {
+                    if (supported) {
+                        this._supportsWebXR = true;
+                        resolve(true);
+                    } else {
+                        this._supportsWebXR = false;
+                        resolve(false);
+                    }
+                }).catch(() => {
+                    this._supportsWebXR = false;
+                    resolve(false);
+                });
+        });
+
+    }
+
+    public static GetSupportsARQuickLook(): boolean {
+        const userAgent = navigator.userAgent;
+
+        // Check if the device is running iOS
+        const isIOS = /iPad|iPhone|iPod/.test(userAgent) && !(window as unknown as Window & { MSStream?: string }).MSStream;
+        if (!isIOS) {
+            return false;
+        }
+
+        // Extract iOS version
+        const match = userAgent.match(/OS (\d+)_/);
+        if (!match || match.length < 2) {
+            return false;
+        }
+        const iOSVersion = parseInt(match[1], 10);
+
+        // Minimum iOS version for QuickLook support
+        const minQuickLookVersion = 12;
+
+        // Check if the iOS version is supported
+        if (iOSVersion < minQuickLookVersion) {
+            return false;
+        }
+
+        // Check for supported browser
+        const isSupportedBrowser = /^((?!chrome|android).)*safari|CriOS|FxiOS/i.test(userAgent);
+        if (isSupportedBrowser) {
+            return true;
+        }
+
+        // Default to false if none of the conditions are met
+        return false;
+    }
+
+    public static get isMobile(): boolean {
```

Comment:
Maybe this way https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent#mobile_device_detection ?

Reply:
Seems to be invalid: 
https://stackoverflow.com/questions/69850527/why-is-navigator-maxtouchpoints-10-in-chrome-on-desktop

Reply:
not sure, there is not to many votes on that one might be an isolated case, also 1 year ago, lets try it on latest chrome it should be fixed as it as a recommended implementation... I would say there more edge cases when implementing it by user agent sniffing

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Navigator/maxTouchPoints Seems to work with every touch screen device.

For example: Is a Microsoft Surface a desktop or a mobile device? What about laptops that have a touch screen? They also most probably have a maxTouchPoints amount of more than 0. But AR might not be supported on them.

Reply:
Maybe some like `if ('ontouchstart' in window) {
  /* browser with Touch Events
     running on touch-capable device */
}` might be better?

Reply:
If there is a mobile device out there (or at least something that is consireded mobile) that does not support touch, you have to check the screen size via `if (window.matchMedia("(max-width: 767px)").matches)
{
// The viewport is less than 768 pixels wide
}`.

Seems kinda weird to me, but still an option as a fallback to the touch check.

Reply:
I think Microsoft Surface is a mobile device including its "tablet mode" but nevertheless you are right, being touchable doesnt mean its mobile. It goes in the same direction to not mix AR with Mobile... also a desktop with a compatible device could do AR. So checking against feature support should be the prefered method as we talk before.  I would add the info like isTouchScreen different to isMobile, and supportWebXR, supportQuickLook, etc... so this flags create specific assumptions by the devs using it

Reply:
regarding max-width that should not be right, as you can rotate mobile to landscape view and the resolution might vary a lot, even exceeding some desktops.

Reply:
Maybe checking max-width and height so we get around different orientations ?

Reply:
we could get it via const portrait = window.matchMedia("(orientation: portrait)").matches; , although thats doesnt cover tablets with the same resolution as desktop devices for example, we fall again into feature detection over assuming it via other characteristics. It has implications on debugging and dev as well... when we do a check on OS/Browser/screensize or smt like this instead of feature, we block the testability of it via desktop browser plugins that simulates device support on desktop while developing.

Reply:
Okay. Since we would use the `isMobile` getter for the QR code stuff, we could replace it with `isARCapable`. This is more feature oriented. Because we don't want to know if we are on a mobile device. Instead we assume that being on a mobile device implicates that it is capable of showing AR content.

In the same way I would remove the code for browser and version detection completely. Just because you are right: We are not thinking feature based here.

What do you guys think about this? Are you fine with the decision?

Reply:
Amazing, yes totally agree 👍